nncf/openvino/quantization/weights_compression.py

```diff
@@ -9,23 +9,326 @@
+    [
+        -1.0,
+        -0.6961928009986877,
+        -0.5250730514526367,
+        -0.39491748809814453,
+        -0.28444138169288635,
+        -0.18477343022823334,
+        -0.09105003625154495,
+        0.0,
+        0.07958029955625534,
+        0.16093020141124725,
+        0.24611230194568634,
+        0.33791524171829224,
+        0.44070982933044434,
+        0.5626170039176941,
+        0.7229568362236023,
+        1.0,
+    ]
+)
+
+
+CENTER_OF_NF4_QUANTILES = np.array(
+    [
+        -0.8480964004993439,
+        -0.6106329262256622,
+        -0.4599952697753906,
+        -0.33967943489551544,
+        -0.23460740596055984,
+        -0.13791173323988914,
+        -0.045525018125772476,
+        0.03979014977812767,
+        0.1202552504837513,
+        0.2035212516784668,
+        0.2920137718319893,
+        0.3893125355243683,
+        0.5016634166240692,
+        0.6427869200706482,
+        0.8614784181118011,
+    ]
+)
+
+
+@dataclass
+class WeightCompressionConfig:
+    """
+    Information on how to compress (quantize) a specific weight.
+
+    :param num_bits: number of bits for storing a single quantized value. 8, by default.
+    :param is_nf4: is NF4 format used for quantization. False, by default.
+    :param group_size: number of weights (e.g. 64 or 128) that are independently quantized. When it equals -1,
+        per-channel quantized is assumed, with group size equals to the size of channel.
+    """
+
+    num_bits: int = 8
+    is_nf4: bool = False
+    group_size: int = -1
+
+
+@dataclass
+class WeightNodeParams:
+    """
+    Information about weight node in the ov.Model that is useful for weight compression.
+
+    :param reduction_axes: Axis or axes along which to reduce (collect) different statistics (e.g. min, max).
+    :param num_weights: number of elements in the weight array.
+    :param fq_name: name for the inserted weight compression operation.
+    :param weight_node: the weight node itself.
+    :param original_weight_dtype: type of elements in the weight array.
+    :param compression_config: configuration of weight compression for the weight node.
+    """
+
+    reduction_axes: Union[int, Tuple[int]]
+    num_weights: int
+    fq_name: str
+    weight_node: ov.Node
+    original_weight_dtype: TWeightType
+    compression_config = WeightCompressionConfig()
+
+
+def _int8_compress(weight: np.array, reduction_axes: List[int]) -> Tuple[np.array, np.array, np.array]:
+    """
+    Do int8 asymmetric weight compression.
+
+    :param weight: Weight array to compress
+    :param reduction_axes: Axis or axes along which to reduce (collect) different statistics (e.g. min, max).
+    :return: compressed weights in unsigned int8, scale and zero point that was used for its quantization.
+    """
+    num_bits = 8
+    level_low = 0
+    level_high = 2**num_bits - 1
+
+    min_values = np.min(weight, axis=reduction_axes, keepdims=True)
+    max_values = np.max(weight, axis=reduction_axes, keepdims=True)
```

Comment:
`np.min` and `np.max` work with `axis: Union[int, Tuple[int]]`

Reply:
thanks, corrected